lib/model/classycat_classify.py

```diff
@@ -0,0 +1,129 @@
+        client = self.get_client()
+        completion = client.chat.completions.create(
+            model=self.model_name,
+            messages=[
+                {"role": "user", "content": task_prompt}
+            ],
+            max_tokens=(max_tokens_per_item * items_count) + 15,
+            temperature=0.5
+        )
+
+        return completion.choices[0].message.content
+
+
+class Model(Model):
+    def __init__(self):
+        super().__init__()
+        self.output_bucket = os.getenv("CLASSYCAT_OUTPUT_BUCKET")
+        self.batch_size_limit = int(os.environ.get("CLASSYCAT_BATCH_SIZE_LIMIT"))
+        llm_model_name = os.environ.get('CLASSYCAT_LLM_MODEL_NAME')
+        self.llm_client = OpenRouterClient(llm_model_name)
+
+
+    def format_input_for_classification_prompt(self, items):
+        return '\n'.join([f"<ITEM_{i}>{item}</ITEM_{i}>" for i, item in enumerate(items)])
+
+    def parse_classification_results(self, raw_results):
+        results = raw_results.strip()
+        results = results.split('<OUTPUT>')[1].split('</OUTPUT>')[0].strip().split('\n')
+
+        final_results = []
+        for i, result in enumerate(results):
+            final_result = result.split(f'</CATEGORIES_{i}>')[0].split(f'<CATEGORIES_{i}>')[1]
+            final_results.append(final_result.strip().split(';'))
+
+        return final_results
+
+    def classify(self, prompt_template, items, max_tokens_per_item=200):
+        task_prompt = prompt_template.replace('<INSERT_ITEMS_HERE>', self.format_input_for_classification_prompt(items))
+        raw_results = self.llm_client.classify(task_prompt, len(items), max_tokens_per_item)
+        logger.info(f"Raw completion: {raw_results}")
+        return self.parse_classification_results(raw_results)
+
+    def classify_and_store_results(self, schema_id, items):
+        # load prompt from schema
+        schema = load_file_from_s3(self.output_bucket, f"{schema_id}.json")
+        schema = json.loads(schema)
+        prompt = schema['prompt']
+        logger.info(f"Prompt for schema {schema['schema_name']} retrieved successfully")
+
+        item_texts = [item['text'] for item in items]
+        classification_results = self.classify(prompt, item_texts)
+
+        if (classification_results is None or len(classification_results) == 0
+                or len(classification_results) != len(items)):
+            logger.info(f"Classification results: {classification_results}")
+            raise Exception(f"Not all items were classified successfully: "
+                            f"input length {len(items)}, output length {len(classification_results)}")
+
+        final_results = [{'id': items[i]['id'], 'text': items[i]['text'], 'labels': classification_results[i]}
+                         for i in range(len(items))]
+        results_file_id = str(uuid.uuid4())
+        upload_file_to_s3(self.output_bucket, f"{schema_id}/{results_file_id}.json", json.dumps(final_results))
```

Comment:
Maybe the file upload should only be enabled in Live?

Reply:
I would like to be able to test that locally as well and make sure the results are being properly stored in minio

Reply:
I think we'd still need to upload in Minio to be fully functional? Is that acceptable or no?

Reply:
I think it is fine to leave it enabled in any of the environments.  But i was thinking it would probably only be the data in live that we'd want to be saving for possible model training?

Reply:
I'm gonna leave it as is for complete testing of functionality in local development